model_compression_toolkit/common/mixed_precision/search_methods/linear_programming.py

```diff
@@ -176,27 +146,43 @@ def _formalize_problem(layer_to_indicator_vars_mapping: Dict[int, Dict[int, LpVa
     # Bound the feasible solution space with the desired KPI.
     # Creates separate constraints for weights KPI and activation KPI.
     if target_kpi is not None:
-        total_weights_consumption = []
-        total_activation_consumption = []
+        indicators = []
         for layer in layer_to_metrics_mapping.keys():
-            if not np.isinf(target_kpi.weights_memory):
-                weights_by_indicators = [indicator * layer_to_kpi_mapping[layer][nbits].weights_memory
-                                         for nbits, indicator in layer_to_indicator_vars_mapping[layer].items()]
-                total_weights_consumption.extend(weights_by_indicators)
-            if not np.isinf(target_kpi.activation_memory):
-                activation_by_indicators = [indicator * layer_to_kpi_mapping[layer][nbits].activation_memory
-                                            for nbits, indicator in layer_to_indicator_vars_mapping[layer].items()]
-                total_activation_consumption.extend(activation_by_indicators)
-
-        # Total model memory size is bounded to the given KPI.
-        # Since total_weights_consumption and total_activation_consumption is the contribution
-        # to the minimal possible KPI (for weights and activation respectively),
-        # we bound the problem by the difference of the target KPI to the minimal KPI.
+            for _, indicator in layer_to_indicator_vars_mapping[layer].items():
+                indicators.append(indicator)
+
+        indicators_arr = np.array(indicators)
+        indicators_matrix = np.diag(indicators_arr)
+
         if not np.isinf(target_kpi.weights_memory):
-            lp_problem += lpSum(total_weights_consumption) <= target_kpi.weights_memory - minimal_kpi.weights_memory
+            weights_kpi_metrix = search_manager.compute_kpi_metrix(KPITarget.WEIGHTS)
+            indicators_weights_kpi_matrix = np.matmul(weights_kpi_metrix, indicators_matrix)
+
+            weights_kpi_sum_vector = [
```

Comment:
Let's add a comment here that the sum is possible due to the diagonal indicators matrix, and this is aimed to reduce the matrix dimensions.